app/internal/corona_stats.py

```diff
@@ -0,0 +1,155 @@
+import json
+import random
+from datetime import date, datetime
+from typing import Any, Dict
+
+import httpx
+from fastapi import Depends
+from loguru import logger
+from sqlalchemy import desc, func
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from sqlalchemy.orm.exc import NoResultFound
+from sqlalchemy.sql.sqltypes import JSON
+
+from app.database.models import CoronaStats
+from app.dependencies import get_db
+
+DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+CORONA_API_URL = (
+    "https://datadashboardapi.health.gov.il/api/queries/vaccinated"
+)
+user_agent_list = [
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5)"
+    " AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.1.1 Safari/605.1.15",
+    "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:77.0)"
+    " Gecko/20100101 Firefox/77.0",
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) "
+    "AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537.36",
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:77.0) "
+    "Gecko/20100101 Firefox/77.0",
+    "Mozilla/5.0 (Windows NT 10.0; Win64; x64) "
+    "AppleWebKit/537.36 (KHTML, like Gecko)"
+    " Chrome/83.0.4103.97 Safari/537.36",
+]
+
+headers = {"User-Agent": random.choice(user_agent_list)}
+
+
+def create_stats_object(corona_stats_data: JSON) -> CoronaStats:
+    """ JSON -> DB Object """
+    return CoronaStats(
+        date_=datetime.strptime(
+            corona_stats_data.get("Day_Date"),
+            DATETIME_FORMAT,
+        ),
+        vaccinated=corona_stats_data.get("vaccinated"),
+        vaccinated_cum=corona_stats_data.get("vaccinated_cum"),
+        vaccinated_population_perc=corona_stats_data.get(
+            "vaccinated_population_perc",
+        ),
+        vaccinated_second_dose=corona_stats_data.get(
+            "vaccinated_seconde_dose",
+        ),
+        vaccinated_second_dose_cum=corona_stats_data.get(
+            "vaccinated_seconde_dose_cum",
+        ),
+        vaccinated_second_dose_perc=corona_stats_data.get(
+            "vaccinated_seconde_dose_population_perc",
+        ),
+    )
+
+
+def serialize_stats(stats_object: CoronaStats) -> Dict:
+    """ DB Object -> Dict """
+    return {
+        "vaccinated_second_dose_perc": (
+            stats_object.vaccinated_second_dose_perc
+        ),
+        "vaccinated_second_dose_cum": (
+            stats_object.vaccinated_second_dose_cum
+        ),
+    }
+
+
+def serialize_dict_stats(stats_dict: Dict) -> Dict:
+    """ api Dit -> pylender Dict """
+    return {
+        "vaccinated_second_dose_perc": (
+            stats_dict.get("vaccinated_seconde_dose_population_perc")
+        ),
+        "vaccinated_second_dose_cum": (
+            stats_dict.get("vaccinated_seconde_dose_cum")
+        ),
+    }
+
+
+def save_corona_stats(
+    corona_stats_data: JSON,
```

Comment:
is the type(corona_stats_data) result of this a JSON object?

Reply:
i dont understant the comment and what i should fix in the code, can you please explain ?

Reply:
In you code do print(type(corona_stats_data)). What type does it show there? I was just wondering if the type is actually JSON and not a Dict.

Reply:
its formmating and saving the json from api to the database

Reply:
Not what I asked...

Reply:
its JSON loaded from the http response